localstack/services/apigateway/apigateway_listener.py

```diff
@@ -118,7 +118,10 @@ def forward_request(self, method, path, data, headers):
                     response = Response()
                     parsed_result = json.loads(result)
                     response.status_code = int(parsed_result['statusCode'])
-                    response.headers.update(parsed_result['headers'])
+                    try:
+                        response.headers.update(parsed_result['headers'])
```

Comment:
@whummer I have implemented your suggestion. Are we sure `response.headers` has that initial value, though?